### PR TITLE
templates/openshift: make the maintenance template generic

### PR DESCRIPTION
**Pull request by croissanne**
We could deploy this job for both composer and each tenant's workers that's present in app-intf. Then we can remove the maintenance bits from the composer template.

### CODE REVIEW

File: templates/openshift/maintenance.yml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-  template: composer-worker-maintenance
+  template: image-builder-maintenance
 objects:
 - apiVersion: batch/v1
   kind: CronJob
   metadata:
     labels:
       service: image-builder
-    name: crc-worker-maintenance
+    name: image-builder-maintenance
```
Comment (schuellerf) [inline templates/openshift/maintenance.yml:20]: also rename this in line 35 ?

**Review by schuellerf — CHANGES_REQUESTED** [review]
Will the DB maintenance then be remove from the composer.yml?

**Comment by croissanne** [comment]
> Will the DB maintenance then be remove from the composer.yml?

exactly, we'll just deploy the maintenance template twice, once for composer and once for the workers.

**Comment by schuellerf** [comment]
probably with
```
- name: maintenance
  url: https://github.com/osbuild/osbuild-composer
  path: /templates/maintenance.yml
  targets:
  - namespace:
      $ref: /services/image-builder/namespaces/workers-stage.yml
    upstream:
      instance:
        $ref: /dependencies/ci-ext/ci-ext.yml
      name: osbuild-osbuild-composer-gh-build-main
    ref: main
    parameters:
      PGSSLMODE: "require"
      MAINTENANCE_DRY_RUN: "true"
      ENABLE_AWS_MAINTENANCE: "true"
      ENABLE_GCP_MAINTENANCE: "true"
>      ENABLE_DB_MAINTENANCE: "false"
  - namespace:
      $ref: /services/image-builder/namespaces/composer-stage.yml
    upstream:
      instance:
        $ref: /dependencies/ci-ext/ci-ext.yml
      name: osbuild-osbuild-composer-gh-build-main
    ref: main
    parameters:
      PGSSLMODE: "require"
      MAINTENANCE_DRY_RUN: "true"
      ENABLE_DB_MAINTENANCE: "true"
```
as ENABLE_DB_MAINTENANCE is mandatory, right? 🤔

**Review by schuellerf — APPROVED** [review]
LGTM

**Comment by croissanne** [comment]
> 
> as ENABLE_DB_MAINTENANCE is mandatory, right? 🤔

yea but the template sets it to false by default (required & default value just prevent it from being set to "")

**Pull request merged**